### PR TITLE
Bug 1814592 - Instrument the time needed for a successful/failure upload of a ping

### PR DESCRIPTION
**Pull request by badboy**
Based on #2332

### CODE REVIEW

**Review by travis79 — APPROVED** [review]

File: glean-core/src/internal_metrics.rs [hunk 1]
```diff
@@ -82,6 +85,18 @@ impl AdditionalMetrics {
                 },
                 None,
             ),
+
+            shutdown_wait: TimingDistributionMetric::new(
```
[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/src/internal_metrics.rs:89]: uff, not easily.
Our internal/core metrics are a mess. Some are global, others are directly attached as part of other structs.
We can file a bug to look into this though.

File: glean-core/metrics.yaml [hunk 2]
```diff
@@ -607,6 +607,36 @@ glean.upload:
...
+      Time needed for a successful send of a Telemetry ping to the servers and getting a reply back
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=TODO
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=TODO
+    data_sensitivity:
+      - technical
+    notification_emails:
+      - glean-team@mozilla.com
+    expires: never
```
[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/metrics.yaml:623]: Riiight, most other metrics in this category don't have one either.
I'm ok adding mine here though, will need to add the data review anyway.

[the remark this answers is not in the document]
Comment (travis79) [reply glean-core/metrics.yaml:623]: Feel free to use mine, I will be responsible for these if you want.

[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/metrics.yaml:623]: muhahaha, nah. in this case it's fine for me. Not much of a burden.

File: glean-core/src/internal_metrics.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (badboy) [reply glean-core/src/internal_metrics.rs:89]: https://bugzilla.mozilla.org/show_bug.cgi?id=1815213

**Pull request merged**